src/en/anchira/src/eu/kanade/tachiyomi/extension/en/anchira/Anchira.kt

```diff
@@ -0,0 +1,240 @@
+                        else -> ""
+                    }
+
+                    if (sort.isNotEmpty()) url.addQueryParameter("sort", sort)
+                    if (filter.state?.ascending == true) url.addQueryParameter("order", "1")
+                }
+
+                else -> {}
+            }
+        }
+
+        return GET(url.toString(), headers)
+    }
+
+    override fun searchMangaParse(response: Response) = latestUpdatesParse(response)
+
+    // Details
+
+    override fun mangaDetailsRequest(manga: SManga) = GET("$apiUrl/${getPathFromUrl(manga.url)}", headers)
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val bytes = response.body.bytes()
+        val data = msgPack.decodeFromByteArray<Entry>(bytes)
+
+        return SManga.create().apply {
+            url = "$baseUrl/g/${data.id}/${data.key}"
+            title = data.title
+            artist = data.tags.filter { it.namespace == 1 }.joinToString(", ") { it.name }
+            author = data.tags.filter { it.namespace == 2 }.joinToString(", ") { it.name }
+            genre = data.tags.filter { it.namespace != 1 && it.namespace != 2 }.joinToString(", ") { it.name }
+            status = SManga.UNKNOWN
+            thumbnail_url = "$cdnUrl/${data.id}/${data.key}/m/${data.data[data.thumbIndex].name}"
+            update_strategy = UpdateStrategy.ONLY_FETCH_ONCE
+        }
+    }
+
+    override fun getMangaUrl(manga: SManga) = manga.url
+
+    // Chapter
+
+    override fun chapterListRequest(manga: SManga) = mangaDetailsRequest(manga)
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val bytes = response.body.bytes()
+        val data = msgPack.decodeFromByteArray<Entry>(bytes)
+
+        return listOf(
+            SChapter.create().apply {
+                url = "$baseUrl/g/${data.id}/${data.key}"
+                name = "Chapter"
+                date_upload = data.publishedAt
+                chapter_number = 1f
+            },
+        )
+    }
+
+    // Page List
+
+    override fun pageListRequest(chapter: SChapter) = GET("$apiUrl/${getPathFromUrl(chapter.url)}}", headers)
+
+    override fun pageListParse(response: Response): List<Page> {
+        val bytes = response.body.bytes()
+        val data = msgPack.decodeFromByteArray<Entry>(bytes)
+
+        return data.data.mapIndexed { i, img -> Page(i, url = "$cdnUrl/${data.id}/${data.key}/${data.hash}/b/${img.name}") }
+    }
+
+    override fun fetchImageUrl(page: Page): Observable<String> {
+        val imageQuality = if (preferences.getBoolean(IMAGE_QUALITY_PREF, false)) "a" else "b"
+
+        return Observable.just(page.url.replace("/b/", "/$imageQuality/"))
+    }
+
+    override fun imageUrlParse(response: Response) = throw UnsupportedOperationException("Not used")
+
+    // Settings
+
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        val imageQualityPref = androidx.preference.SwitchPreferenceCompat(screen.context).apply {
```

Comment:
```suggestion
        val imageQualityPref = SwitchPreferenceCompat(screen.context).apply {
```
import `androidx.preference.SwitchPreferenceCompat`